docs/source/user_guide/code-snippets.md

```diff
@@ -21,62 +21,151 @@ The ability to reuse pieces of code allows users to avoid doing repetitive work,
+1. Select the `Code Snippets` tab from the JupyterLab sidebar.
+1. Click the trash can next to the code snippet entry.
+
+#### Using code snippets
+
+To search for a code snippet, type a keyword in the search bar or select one or more tags.
+
+  ![Search Code Snippets](../images/search-code-snippets.png)
 
-To add a code snippet, simply click the `+` button in the top right corner of the code snippets side pane, and an editor will appear in the main area of JupyterLab.
+To insert the code snippet content into an open editor, click the insert button or use drag and drop. 
 
-![Add Code Snippets](../images/add-code-snippets.png)
+  ![Insert Code Snippet](../images/insert-code-snippet.png)
 
-To edit a code snippet, click the edit button to open an editor in the main area of JupyterLab.
+Code snippets are automatically added as code blocks in markdown files and notebook markdown cells.
+When inserting snippets into executable editors (e.g. a notebook code cell or a Python/R file editor), the extension will verify kernel language compatibility, warning the user when a mismatch is detected.
+
+### Managing code snippets using the Elyra CLI
 
-![Edit Code Snippets](../images/edit-code-snippets.png)
+You can list, create, edit, or delete code snippets using the `elyra-metadata` CLI.
 
-Click "Save and Close" to see your edits in the code snippets side pane.
+#### Listing code snippets
 
-### Finding Code Snippets
+To list code snippets run
 
-To search for a code snippet, type a keyword in the search bar. Code snippets can also be filtered by selecting tags when clicking on the tag icon.
+```
+elyra-metadata list code-snippets
+```
 
-![Search Code Snippets](../images/search-code-snippets.png)
+The output lists for each code snippet the name and the name of the associated JSON formatted metadata file, which is stored in the JupyterLab data directory in the `metadata/code-snippets` subdirectory.
 
-#### Code Snippet data
-This extension uses [Elyra Metadata Service](../developer_guide/metadata) and in addition to the frontend editor detailed above, this can also be configured on the backend.
-To configure metadata for code snippets through the backend, locate `[JUPYTER DATA DIR]/metadata/` folder and create the subdirectory `code-snippets/`.
-This is where code snippet json files are stored, following a schema defined in
-[code snippets metadata](https://github.com/elyra-ai/elyra/blob/master/elyra/metadata/schemas/code-snippet.json)
+```
+Available metadata instances for code-snippets (includes invalid):
 
-This is a sample json file:
-```json
-{
-	"schema_name": "code-snippet",
-	"display_name": "is_even",
-	"metadata": {
-		"description": "Check if number is even",
-		"language": "python",
-		"code": [
-			"def is_even(num):",
-			"   return num % 2 == 0"
-		],
-        "tags": [
-		  "python"
-        ]
-	}
-}
+Schema   	Instance		Resource  
+---------   	--------		--------
+code-snippet   	preview_dataframe	/.../metadata/preview_dataframe.json 
```

Comment:
shouldn't the path be `/.../metadata/code-snippets/preview_dataframe.json ` ?

Reply:
jap